### DISCUSSION TITLE
Custom folder structure config

### DISCUSSION THREAD

**Discussion by brownphotographic** [photoprism/photoprism#4501] [Q&A] [answered]
Hi

While it is possible to set a custom folder for images to be stored, it appears that PhotoPrism doesn't allow for configuring the folder structure itself.

What I am trying to do:
I have a large photo collection of originals that I would like to maintain, and then just point PhotoPrism at that. Is it possible to do this via a config change? I couldn't see any references in the documentation.

What I believe PhotoPrism assumes:
- User maintains photo library elsewhere.
- User imports images into PhotoPrism
- PhotoPrism makes a copy of those images in its own folder format

**Comment by graciousgrey** [comment, marked as answer]
Hey,

The import feature that moves/copies  your files from the import to the originals folder  in a year/month structure is optional. 
Alternatively you can let PhotoPrism index your existing library:

- https://docs.photoprism.app/user-guide/library/
- https://docs.photoprism.app/getting-started/docker-compose/#photoprismoriginals

For this you just need to configure your existing library as originals folder.

**Reply by brownphotographic** [reply]
Thank you!
